src/utils.rs

```diff
@@ -116,6 +116,38 @@ impl From<NibbleKey> for ByteKey {
     }
 }
 
+pub fn remove_indicator_prefix(key_nibbles: Vec<u8>) -> Vec<u8> {
+    // only leaf nodes and extension nodes have an indicator prefix
+    if key_nibbles[0] == 1 || key_nibbles[0] == 3 {
+        // if indicator byte is 1, it's an odd-length non-terminal node (extension node)
+        // if 3, it's an odd-length terminal node (leaf node)
+        key_nibbles[1..].to_vec()
+    } else {
+        // if even, should prefix should be either 0 or 2
+        key_nibbles[2..].to_vec()
+    }
+}
+
+pub fn add_indicator_prefix(key_nibbles: Vec<u8>, is_terminator: bool) -> Vec<u8> {
```

Comment:
`NibbleKey` should be used instead of `Vec<u8>` here, to make sure that no bytes are being passed.